Read IAM resource name from install-config directly in post-destroy check.

yunjiang29 opened the pull request:
* Fix tag [checks error](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/55076/rehearse-55076-periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-aws-ipi-byo-iam-role-arm-f2/1819208608185847808)
```
FAIL: tag check ci-op-t7tmvpb6-c794e-byo-role-master: kubernetes.io/cluster/ci-op-t7tmvpb6-c794e-b777l tag was attactehd
```
* Read the IAM resource name from install-config directly instead of from instances that have been deleted while running this step
* Re-format bash script.

yunjiang29 commented:
@jianlinliu PATL

jianlinliu commented on ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-commands.sh:
```diff
@@ -10,115 +10,79 @@ trap 'delete_all' EXIT TERM INT
-	aws --region $REGION iam get-role --role-name ${worker_role}
-	aws --region $REGION iam get-policy --policy-arn ${worker_policy_arn}
-fi
+echo "Deleting BYO-IAM resources ..."
```
So we will can not use this step to test if the BYO-iam-profile exists or not after clusters get destroyed.

yunjiang29 replied:
That's what I realized after asking you for approval. 
Fixed now, read IAM name from SHARED_DIR directly.
Thanks

jianlinliu commented on ci-operator/step-registry/aws/deprovision/iam-role/aws-deprovision-iam-role-commands.sh:
```diff
@@ -10,92 +10,55 @@ trap 'delete_all' EXIT TERM INT
-    aws --region $REGION iam get-role --role-name ${worker_role}
-    aws --region $REGION iam get-policy --policy-arn ${worker_policy_arn}
-fi
+echo "Deleting BYO-IAM resources ..."
```
Maybe we should move `delete_all` from `TRAP` line here ?

yunjiang29 commented:
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-custom-iam-profile

jianlinliu commented on ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-commands.sh:
```diff
@@ -10,115 +10,112 @@ trap 'delete_all' EXIT TERM INT
-	aws --region $REGION iam get-role --role-name ${worker_role}
-	aws --region $REGION iam get-policy --policy-arn ${master_policy_arn}
-	aws --region $REGION iam get-policy --policy-arn ${worker_policy_arn}
+  aws --region $REGION iam get-instance-profile --instance-profile-name "$(head -n 1 ${SHARED_DIR}/aws_byo_profile_name_master)"
```
Can we directly use `ic_platform_profile` instead of getting it from file ?

yunjiang29 replied:
Updated

jianlinliu commented on ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-commands.sh:
```diff
@@ -10,115 +10,112 @@ trap 'delete_all' EXIT TERM INT
+  aws --region $REGION iam get-instance-profile --instance-profile-name "$(head -n 1 ${SHARED_DIR}/aws_byo_profile_name_worker)"
+  aws --region $REGION iam get-role --role-name "$(head -n 1 ${SHARED_DIR}/aws_byo_role_name_master)"
+  aws --region $REGION iam get-role --role-name "$(head -n 1 ${SHARED_DIR}/aws_byo_role_name_worker)"
+  aws --region $REGION iam get-policy --policy-arn "$(head -n 1 ${SHARED_DIR}/aws_byo_policy_arn_master)"
```
Is there any step to create `${SHARED_DIR}/aws_byo_policy_arn_master`, I found nothing.

yunjiang29 replied:
https://github.com/openshift/release/blob/master/ci-operator/step-registry/aws/provision/iam-profile/aws-provision-iam-profile-commands.sh#L104

jianlinliu replied:
ah, get it.

yunjiang29 commented:
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-custom-iam-profile

yunjiang29 commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-aws-ipi-byo-iam-role-arm-f2

jianlinliu commented on ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-commands.sh:
```diff
@@ -10,115 +10,105 @@ trap 'delete_all' EXIT TERM INT
+  # detach role
+  attached_roles=$(aws --region $aws_region iam get-instance-profile --instance-profile-name ${profile_name} | jq -r .InstanceProfile.Roles[].RoleName)
+  for role_name in $attached_roles; do
+    if [ X"$role_name" == X"" ]; then
```
In the `for` loop, "role_name" would never be empty, right, sounds like we do not need this if condition.

yunjiang29 replied:
Yes, but this piece of code is very old, I don't remember if I met the output format exceptions that caused `role_name` is empty.

jianlinliu replied:
@yunjiang29 do you mind we have a try to remove the `if` condition to see what happen ?

yunjiang29 replied:
OK

jianlinliu commented on ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-commands.sh:
```diff
@@ -10,115 +10,105 @@ trap 'delete_all' EXIT TERM INT
-	aws --region $REGION iam get-role --role-name ${worker_role}
-	aws --region $REGION iam get-policy --policy-arn ${master_policy_arn}
-	aws --region $REGION iam get-policy --policy-arn ${worker_policy_arn}
+  aws --region $REGION iam get-instance-profile --instance-profile-name "${ic_platform_profile}"
```
One more questions, if profile is already deleted, the `ddelete_all` in TRAP line would delete it again, that function would not exit immediately because the target profile is not found, right?

yunjiang29 replied:
`delete_all` will always be executed, but I think depends on `set -o errexit`, it may fail inside `delete_all`:  https://github.com/yunjiang29/release/blob/619687f362027f3070ad1cb0bf9e32f2ceceed3e/ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-commands.sh#L60

jianlinliu replied:
> depends on set -o errexit, it may fail inside delete_all

Yeah, that is my concern, if it is, maybe we need to `set +e` inside `delete_all` to avoid the unexpected exit.

yunjiang29 commented:
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-custom-iam-profile

yunjiang29 commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-aws-ipi-byo-iam-role-arm-f2

yunjiang29 commented:
`aws-deprovision-iam-profile`
https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/55076/rehearse-55076-periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-aws-ipi-byo-iam-role-arm-f2/1819208608185847808/artifacts/aws-ipi-byo-iam-role-arm-f2/aws-deprovision-iam-role/build-log.txt
```
Deleting BYO-IAM resources ...
Deleteing role: ci-op-t7tmvpb6-c794e-byo-role-master
	Detaching policy: arn:aws:iam::301721915996:policy/ci-op-t7tmvpb6-c794e-byo-policy-master
	Deleted.
Deleteing role: ci-op-t7tmvpb6-c794e-byo-role-worker
	Detaching policy: arn:aws:iam::301721915996:policy/ci-op-t7tmvpb6-c794e-byo-policy-worker
	Deleted.
Deleting policy: arn:aws:iam::301721915996:policy/ci-op-t7tmvpb6-c794e-byo-policy-master
	Deleted.
Deleting policy: arn:aws:iam::301721915996:policy/ci-op-t7tmvpb6-c794e-byo-policy-worker
	Deleted.
```

jianlinliu commented:
/lgtm

yunjiang29 commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-aws-ipi-byo-iam-role-arm-f2

yunjiang29 commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-multi-nightly-aws-ipi-byo-iam-role-fips-amd-f28-destructive

yunjiang29 commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-multi-nightly-aws-ipi-byo-iam-role-arm-f2

yunjiang29 commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-multi-nightly-aws-ipi-byo-iam-role-fips-amd-f28-destructive

jianlinliu commented on ci-operator/step-registry/cucushift/installer/check/aws/byo-iam-role/cucushift-installer-check-aws-byo-iam-role-commands.sh:
```diff
@@ -106,22 +106,66 @@ else
+# for 4.15 and below, no tag is attached to BYO-Role
+
+function has_tags() {
+  local txt=$1
```
```suggestion
  local txt="$1"
```

jianlinliu commented on ci-operator/step-registry/cucushift/installer/check/aws/byo-iam-role/cucushift-installer-check-aws-byo-iam-role-commands.sh:
```diff
@@ -106,22 +106,66 @@ else
+
+function has_tags() {
+  local txt=$1
+  if grep -iE "TAGS.*kubernetes.io/cluster/${INFRA_ID}" $txt; then
```
```suggestion
  if grep -iE "TAGS.*kubernetes.io/cluster/${INFRA_ID}" "$txt"; then
```

yunjiang29 commented:
/pj-rehearse ack

jianlinliu commented:
/lgtm

yunjiang29 commented:
/pj-rehearse ack
